projects/plugins/boost/app/admin/class-regenerate-admin-notice.php

```diff
@@ -21,54 +22,78 @@ public static function enable() {
 	}
 
 	public static function dismiss() {
-		Transient::delete( 'regenerate_admin_notice', false );
+		Transient::delete( 'regenerate_admin_notice' );
 	}
 
 	public static function is_enabled() {
 		return Transient::get( 'regenerate_admin_notice', false );
 	}
 
+	public static function enable_suggestion() {
+		Transient::set( 'regenerate_admin_suggestion', true );
+	}
+
+	public static function dismiss_suggestion() {
+		Transient::delete( 'regenerate_admin_suggestion' );
+	}
+
+	public static function is_suggestion_enabled() {
+		return Transient::get( 'regenerate_admin_suggestion', false );
+	}
```

Comment:
Is there a reason to use a separate transient for suggestions, and not just store a different value in the same transient for suggestions and notices?

Reply:
Yes, one flag determines if there has been a change since the last C.CSS change, one that determines if the dialog was dismissed within the last 7 days(by dismiss button or fresh C.CSS). I guess I need to refactor it as I now feel like the logic is a bit confusing, but I don't think it can be tied to the C.CSS notice.

Reply:
I have made the changes I was thinking about. The commit is f473f19.

To explain, we need two flags:
1. Marking there has been a HTML/CSS change since last generation
2. Marking that the notice should be dismissed for a week

Before f473f19, it was counter-intuitive. Flag#1 was being handled from `Regenerate_Admin_Notice`, and Flag#2 was being handled by `Critical_CSS_State`. I have reversed it now and the code should be more self-explanatory.